src/pages/workspace/WorkspaceProfileAddressPage.tsx

```diff
@@ -0,0 +1,101 @@
+
+type WorkspaceProfileAddressPagePolicyProps = WithPolicyProps;
+
+type WorkspaceProfileAddressPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.ADDRESS> & WorkspaceProfileAddressPagePolicyProps;
+
+function WorkspaceProfileAddressPage({policy}: WorkspaceProfileAddressPageProps) {
+    const {translate} = useLocalize();
+    const address = useMemo(() => policy?.address, [policy]);
+    const [currentCountry, setCurrentCountry] = useState(address?.country);
+    const [street1, street2] = (address?.addressStreet ?? '').split('\n');
+    const [state, setState] = useState(address?.state);
+    const [city, setCity] = useState(address?.city);
+    const [zipcode, setZipcode] = useState(address?.zip);
+
+    const updatePolicyAddress = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.HOME_ADDRESS_FORM>) => {
+        if (!policy) {
+            return;
+        }
+        updateAddress(policy?.id, {
+            addressStreet: `${values.addressLine1?.trim() ?? ''}\n${values.addressLine2?.trim() ?? ''}`,
+            city: values.city.trim(),
+            state: values.state.trim(),
+            zip: values?.zipPostCode?.trim().toUpperCase() ?? '',
+            country: values.country,
+        });
+        Navigation.goBack();
+    };
+
+    const handleAddressChange = useCallback((value: unknown, key: unknown) => {
+        const countryValue = value as Country | '';
+        const addressKey = key as keyof CompanyAddress;
+
+        if (addressKey !== 'country' && addressKey !== 'state' && addressKey !== 'city' && addressKey !== 'zip') {
+            return;
+        }
+        if (addressKey === 'country') {
+            setCurrentCountry(countryValue);
+            setState('');
+            setCity('');
+            setZipcode('');
+            return;
+        }
+        if (addressKey === 'state') {
+            setState(countryValue);
+            setCity('');
+            setZipcode('');
+            return;
+        }
+        if (addressKey === 'city') {
+            setCity(countryValue);
+            setZipcode('');
+            return;
+        }
+        setZipcode(countryValue);
+    }, []);
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            testID={WorkspaceProfileAddressPage.displayName}
+        >
+            <HeaderWithBackButton
+                title={translate('common.companyAddress')}
+                shouldShowBackButton
+                onBackButtonPress={() => Navigation.goBack()}
+            />
+            <AddressForm
+                formID={ONYXKEYS.FORMS.HOME_ADDRESS_FORM}
+                onSubmit={updatePolicyAddress}
+                submitButtonText={translate('common.save')}
+                city={city}
+                country={currentCountry}
+                onAddressChanged={handleAddressChange}
+                state={state}
+                street1={street1}
+                street2={street2}
+                zip={zipcode}
```

Comment:
afterwards, you can pass form data as a whole, or include it inside of AddressForm

Reply:
Same as [here](https://github.com/Expensify/App/pull/38381#discussion_r1580776921).